common/djangoapps/embargo/models.py

```diff
@@ -123,9 +129,50 @@ class RestrictedCourse(models.Model):
         help_text=ugettext_lazy(u"The message to show when a user is blocked from accessing a course.")
     )
 
+    @classmethod
+    def cache_key_name(cls):
+        """Return the name of the key to use to cache the current restricted course list"""
+        return 'embargo/RestrictedCourse/courses'
+
+    @classmethod
+    def is_restricted_course(cls, course_id):
+        """
+        Check if the course is in restricted list
+
+        Args:
+            course_id (str): course_id to look for
+
+        Returns:
+            Boolean
+            True if course is in restricted course list.
+        """
+        return unicode(course_id) in cls._get_restricted_courses_from_cache()
+
+    @classmethod
+    def _get_restricted_courses_from_cache(cls):
+        """
+        Cache all restricted courses and returns the list of course_keys that are restricted
+        """
+        restricted_countries = cache.get(cls.cache_key_name())
+        if not restricted_countries:
+            restricted_countries = list(RestrictedCourse.objects.values_list('course_key', flat=True))
+            cache.set(cls.cache_key_name(), restricted_countries)
```

Comment:
@awais786 One small thing I noticed just now: you're calling the variable `restricted_countries` but it should be `restricted_courses`.  Could you please make this change?